File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.py
from collections import defaultdict
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = defaultdict(int)
        for i in nums:
            counts[i] += 1

        heap = []
        for key, value in counts.items():
            heapq.heappush(heap, (-value, key))

        result = []
        for i in range(0, k):
            result.append(heapq.heappop(heap)[1])

        return result
        
# [1, 2, 2, 3, 3, 3], k = 2
# 1 --> 1
# 2 --> 2
# 3 --> 3

#  3
# 2 1
# pop() --> 3, pop() --> 2

File: Data Structures & Algorithms/duplicate-integer/submission-5.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::set<int> s{};

        for (int &i: nums) {
            if (s.contains(i)) {
                return true;
            }

            s.insert(i);
        }

        return false;
    }
};